cider/exec/plan/validator/SingleNodeValidator.cpp

```diff
@@ -84,6 +84,8 @@ std::vector<substrait::Type> SingleNodeValidator::getRelOutputTypes(
               output_types.emplace_back(getExprOutputType(s_expr, input_types));
             } else {
               CHECK(s_expr.has_selection());
+              auto type = input_types
```

Comment:
```
auto index = s_expr.selection().direct_reference().struct_field().field();
auto type = input_types[index];
output_types.emplace_back(type);
```

Reply:
Maybe current is better to avoid tmp variables?